storage-sqlx/tests/purge.rs

```diff
@@ -0,0 +1,183 @@
+use event_sauce::{
+    prelude::*, AggregateCreate, AggregatePurge, DBEvent, Entity, Event, Persistable,
+    PurgeBuilderExecute, PurgeEntityBuilder,
+};
+use event_sauce_storage_sqlx::{SqlxPgStore, SqlxPgStoreTransaction};
+// use event_sauce::UpdateEntity;
+use sqlx::{postgres::PgQueryAs, PgPool};
+use std::convert::TryFrom;
+use uuid::Uuid;
+
+#[derive(
+    serde_derive::Serialize,
+    serde_derive::Deserialize,
+    sqlx::FromRow,
+    event_sauce_derive::Entity,
+    PartialEq,
+    Debug,
+)]
+#[event_sauce(entity_name = "crud_test_users")]
+struct User {
+    #[event_sauce(id)]
+    id: Uuid,
+    name: String,
+    email: String,
+}
+
+/// The event used to create users in this test suite
+#[derive(
+    serde_derive::Serialize, serde_derive::Deserialize, event_sauce_derive::CreateEventData,
+)]
+#[event_sauce(User)]
+struct UserCreated {
+    name: String,
+    email: String,
+}
+
+impl AggregateCreate<UserCreated> for User {
+    type Error = &'static str;
+
+    fn try_aggregate_create(event: &Event<UserCreated>) -> Result<Self, Self::Error> {
+        let data = event
+            .data
+            .as_ref()
+            .ok_or("Event data must be populated to create User from UserCreated event")?;
+
+        Ok(User {
+            id: event.entity_id,
+            name: data.name.clone(),
+            email: data.email.clone(),
+        })
+    }
+}
+
+/// Event used to purge users in this test suite.
+#[derive(serde_derive::Serialize, serde_derive::Deserialize, event_sauce_derive::PurgeEventData)]
+#[event_sauce(User)]
+struct UserPurged;
+
+impl AggregatePurge<UserPurged> for User {
+    type Error = ();
```

Comment:
This might be better off as `sqlx::Error`. It doesn't really matter in this test, but signals to people reading this test as a reference to implement their own code that the operation can fail with a DB error.

```suggestion
    type Error = sqlx::Error;
```

Reply:
Actually, I don't even think we need the error type. This is a marker-trait.

Reply:
I removed the AggregatePurge trait, we don't need it. The `PurgeEntityBuilder` does not need to aggregate to provide a builder for the PurgeEvent.

Reply:
This reminds me that the names and docs we use are not great. It could use some rethinking, mainly for the Builder generators and their methods. But doing those changes may require refactoring some of our app. I may add a ticket for if you agree.

Reply:
> I may add a ticket for if you agree.

Sure, but a GH issue on this repo not Jira lol.